File: checks/python/src/section508/result.py
from dataclasses import dataclass, field


@dataclass
class Failure:
    id: str
    title: str
    details: list[str] = field(default_factory=list)


@dataclass
class Result:
    ok: bool
    failures: list[Failure] = field(default_factory=list)

    def report(self) -> str:
        lines = []
        for failure in self.failures:
            lines.append(f"{failure.id}: {failure.title}")
            for detail in failure.details:
                lines.append(f"  {detail}")
        return "\n".join(lines)

File: checks/ruby/lib/section508/result.rb
# frozen_string_literal: true

module Section508
  Failure = Struct.new(:id, :title, :details, keyword_init: true)

  class Result
    attr_reader :failures

    def initialize(failures)
      @failures = Array(failures)
    end

    def ok?
      failures.empty?
    end

    alias ok ok?
  end
end
